### PR TITLE
global: core modules stabilization preparations

### PR DESCRIPTION
**Pull request by lnielsen**
* Externalizes requirements.py into requirements-builder package.

* Adds default pytest configuration.

* Integrates isort and check-manifest into TravisCI tests.

* Adds new standard contributing guide.

* Updates standard documentation structure.

* Changes default documentation platform to pythonhosted.org and
  changes Sphinx theme to alablaster including "Fork me" ribbon.

* Removes ultramock from Sphinx configuration.

* Fixes Pytest 2.8.0 issues.

* Adds Python 3 support by default.

* Removes tox.

Signed-off-by: Lars Holm Nielsen <lars.holm.nielsen@cern.ch>

### CODE REVIEW

File: {{ cookiecutter.project_shortname }}/CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
+============
```
Comment (jirikuncar) [inline {{ cookiecutter.project_shortname }}/CONTRIBUTING.rst]: I'm not sure about the header level here

**Comment by jirikuncar** [comment]
If you are sure about https://github.com/inveniosoftware/cookiecutter-invenio-module/pull/25#discussion_r41414722 you can :shipit:

**Comment by lnielsen** [comment]
@jirikuncar Yep you're right.

File: {{ cookiecutter.project_shortname }}/CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
+Contributing
+============
+
+Contributions are welcome, and they are greatly appreciated! Every
+little bit helps, and credit will always be given.
+
+Types of Contributions
+----------------------
```
Comment (tiborsimko) [inline {{ cookiecutter.project_shortname }}/CONTRIBUTING.rst]: I think the CONTRIBUTING guide should be rather simple, kind of like the usual:

- https://github.com/inveniosoftware/claimstore/blob/master/CONTRIBUTING.rst
- https://github.com/cernopendata/opendata.cern.ch/blob/master/CONTRIBUTING.rst

and all the git etc details should rather point to the common documentation located on RTFD.  (coming with https://github.com/inveniosoftware/invenio/issues/1799)

**Pull request merged**